spectral_cube/tests/test_masks.py

```diff
@@ -256,6 +256,30 @@ def test_wcs_validity_check():
     # just checking that this works, not that it does anything in particular
     moment_map = s3.moment(order=1)
 
+def test_wcs_validity_check_failure():
+    cube, data = cube_and_raw('adv.fits')
+    wcs2 = cube.wcs.copy()
+    # add some difference in the 4th decimal place
+    wcs2.wcs.crval[2] += 0.00001
+
+    # can make a mask
+    mask = BooleanArrayMask(data>0, cube._wcs)
+
+    # but if it's not exactly equal, an error should be raised at this step
+    with pytest.raises(ValueError) as exc:
+        cube = cube.with_mask(mask)
+    assert exc.value.args[0] == "WCS does not match mask WCS"
+
+    # this one should work though
+    cube = cube.with_mask(mask, wcs_tolerance=1e-4)
+    assert cube._wcs_tolerance == 1e-4
```

Comment:
In the above examples, you have 3 different cube objects in each case.  I advocate instead:

```
masked_cube1 = cube.with_mask(mask, wcs_tolerance=1e-4)
masked_cube2 = cube.with_mask(mask, wcs_tolerance=1e-6)
```

However, in the case that you want to keep adding masks, you may have to increase the tolerance...
```
masked_cube1 = cube.with_mask(mask, wcs_tolerance=1e-6)
masked_cube2 = masked_cube1.with_mask(mask2, wcs_tolerance=1e-5)
```

Does this make sense?

I'm trying to treat `cube` objects as immutable as much as possible